modules/yandexAnalyticsAdapter.js

```diff
@@ -0,0 +1,222 @@
+import buildAdapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import adapterManager from '../src/adapterManager.js';
+import { logError, logInfo } from '../src/utils.js';
+import CONSTANTS from '../src/constants.json';
+import * as events from '../src/events.js';
+
+const tagTlds = ['com', 'ru'];
+const fileNames = ['tag', 'int']
+const tagUrlTemplate = 'https://mc.yandex.{tld}/metrika/{file}.js';
+export const tagURLs = tagTlds.flatMap((tld) => {
+  const partialTemplate = tagUrlTemplate.replace('{tld}', tld);
+  return fileNames.map((file) => partialTemplate.replace('{file}', file));
+});
+
+const timeoutIds = {};
+const tryUntil = (operationId, conditionCb, cb) => {
+  if (!conditionCb()) {
+    cb();
+    timeoutIds[operationId] = setTimeout(
+      () => tryUntil(conditionCb, conditionCb, cb),
+      100
+    );
+  }
+};
+const waitFor = (operationId, conditionCb, cb) => {
+  if (conditionCb()) {
+    cb();
+  } else {
+    timeoutIds[operationId] = setTimeout(() => waitFor(conditionCb, cb), 100);
+  }
+};
+
+const clearWaitForTimeouts = (timeouts) => {
+  timeouts.forEach((timeoutID) => {
+    if (timeoutIds[timeoutID]) {
+      clearTimeout(timeoutIds[timeoutID]);
+    }
+  });
+};
+
+const SEND_EVENTS_BUNDLE_TIMEOUT = 1500;
+const {
+  EVENTS: {
+    BID_REQUESTED,
+    BID_RESPONSE,
+    BID_ADJUSTMENT,
+    BID_WON,
+    BIDDER_DONE,
+    AUCTION_END,
+    BID_TIMEOUT,
+  },
+} = CONSTANTS;
```

Comment:
Does this declaration work? I get a type error when I try this.

Reply:
Yes, it works, in fact it passes autotests + has been tested manually. Please share error text with me, so we can debug it.

Reply:
Uncaught TypeError: Cannot read properties of undefined (reading 'BID_REQUESTED')
    at <anonymous>:3:5
    
    You cannot nest destructuring assignment. You need to removed the EVENTS object layer.

Reply:
Ok, thanks for improvement, done.